posydon/interpolation/IF_interpolation.py

```diff
@@ -467,7 +471,7 @@ def __init__(self, grid=None, in_keys=None, out_keys=None, in_scaling=None,
             self.interp_in_q = self._interpIn_q(grid)
             self.XT, self.YT = self._grid2array(grid)
             self.valid = self._setValid(
-                grid.final_values[self.c_key], self.XT)
+                grid.final_values["interpolation_class"], self.XT)
```

Comment:
Could you please comment, why we need to have here a hard coded "interpolation class"? For me it feels in this way you make the use case of c_key obsolete. @philipp-rajah, could you please check this change in detail?

Reply:
@ZepeiX This is a good explanation of what's going on.

Was there a change in how we process non-valid models?
As in, originally we used to exclude them in the parameter calculation, while now we run them through it?
As such, we get final values for non-valid models?